app/src/main/java/me/phh/treble/app/HuaweiAudio.kt

```diff
@@ -0,0 +1,46 @@
+package me.phh.treble.app
+
+import android.content.BroadcastReceiver
+import android.content.Context
+import android.content.Intent
+import android.content.IntentFilter
+import android.media.AudioManager
+import android.util.Log
+
+class HuaweiAudio : EntryStartup, BroadcastReceiver() {
+    private var audioManager: AudioManager? = null
+    override fun onReceive(context: Context, intent: Intent) {
+        if (intent.action != Intent.ACTION_HEADSET_PLUG)
+            return
+        if (audioManager == null) {
+            audioManager = context.getSystemService(Context.AUDIO_SERVICE) as AudioManager
+        }
+        val state = intent.getIntExtra("state", -1)
+        when(state) {
```

Comment:
If I'm not mistaken, those should be AudioSystem.DEVICE_STATE_AVAILABLE // AudioSystem.DEVICE_STATE_UNAVAILABLE ?

Reply:
Yep